include/nbl/video/IGPUFence.h

```diff
@@ -31,6 +33,94 @@ class IGPUFence : public core::IReferenceCounted, public IBackendObject
     virtual ~IGPUFence() = default;
 };
 
+class GPUEventWrapper : public core::Uncopyable
```

Comment:
this needs to go to a different file and be done in 3 distinct flavours:
- fence
- semaphore
- polymorphic either fence or semaphore

Reply:
eh okay it doesnt.

Timeline sempahore we dont support.

Events suck because they dont ensure anything can be deleted safely
https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#fundamentals-objectmodel-lifetime

Reply:
it should probably change its name to `GPUFenceEventWrapper`